magenta-lib/src/test/scala/magenta/deployment_type/CloudFormationTest.scala

```diff
@@ -21,32 +21,30 @@ class CloudFormationTest extends FlatSpec with Matchers {
     val cfnStackName = s"cfn-app-PROD"
     val p = DeploymentPackage("app", app, data, "cloudformation", new File("/tmp/packages/webapp"))
 
-    CloudFormation.perAppActions("updateStack")(p)(lookupEmpty, parameters(), stack) should be (List(
-      UpdateCloudFormationTask(
-        cfnStackName,
-        Path(p.srcDir) \ Path.fromString("""cloud-formation/cfn.json"""),
-        Map.empty,
-        PROD,
-        NamedStack("cfn"),
-        true
-      ),
-      CheckUpdateEventsTask(cfnStackName)
-    ))
+    inside(CloudFormation.perAppActions("updateStack")(p)(lookupEmpty, parameters(), stack)) {
+      case List(updateTask, checkTask) =>
+        inside(updateTask) {
+          case UpdateCloudFormationTask(stackName, path, userParams, amiParam, amiTags, _, stage, stack, ifAbsent) =>
+            stackName should be(cfnStackName)
+            path should be(Path(p.srcDir) \ Path.fromString( """cloud-formation/cfn.json"""))
+            userParams should be(Map.empty)
+            amiParam should be("AMI")
+            amiTags should be(Map.empty)
+            stage should be(PROD)
+            stack should be(NamedStack("cfn"))
+            ifAbsent should be(true)
```

Comment:
scalatest has a nice have syntax which you may prefer? It let's you check multiple properties on a thing at once. It's a bit less typesafe (because have takes symbols which it uses to look up the value) but I tend to use it for testing multiple properties on a case class because in tests clarity of meaning is the most important thing.

```scala
inside(updateTask) { ufct =>
  ucft should have (
    'stackName (cfnStackName),
    'propertyName (value)
    // etc
  )
}
```